smart_campus/app/views.py

```diff
@@ -30,17 +32,20 @@ def signup(request):
     """
     email = request.POST.get('email')
     password = request.POST.get('password')
-    nickname = request.POST.get('nickname')
+    nickname = request.POST.get('nickname', '')
+
+    if not email or not password:
+        return HttpResponse('No email or password data', status=400)
 
     if User.objects.filter(email=email).exists():
-        return JsonResponse({'status': 'false', 'message': 'User already exists!'})
+        return HttpResponse('The email is already taken, try another!', status=400)
 
     try:
         User.objects.create_user(email, password, nickname)
     except ValueError as error:
-        return JsonResponse({'status': 'false', 'message': error})
+        return HttpResponse('Email address not valid!', status=400)
```

Comment:
Are you sure that only email can throw exceptions?
If so, the message should be 'Invalid email address.'